src/components/Table.jsx

```diff
@@ -0,0 +1,238 @@
+  CATEGORY: 'CATEGORY',
+  USER: 'USER',
+};
+
+const sorter = (products, query) => {
+  const sortedProducts = structuredClone(products);
+
+  if (!Object.keys(query).length) {
+    return sortedProducts;
+  }
+
+  const [property] = Object.entries(query);
+  const [key, value] = property;
+
+  switch (key) {
+    case QUERY.ID:
+      sortedProducts.sort((a, b) => (
+        value === 'ASC'
+          ? a.id - b.id
+          : b.id - a.id
+      ));
+      break;
+
+    case QUERY.USER:
+      sortedProducts.sort((a, b) => (
+        value === 'ASC'
+          ? a.user.name.localeCompare(b.user.name)
+          : b.user.name.localeCompare(a.user.name)
+      ));
+      break;
+
+    case QUERY.CATEGORY:
+      sortedProducts.sort((a, b) => (
+        value === 'ASC'
+          ? a.category.title.localeCompare(b.category.title)
+          : b.category.title.localeCompare(a.category.title)
+      ));
+      break;
+
+    case QUERY.PRODUCT:
+      sortedProducts.sort((a, b) => (
+        value === 'ASC'
+          ? a.name.localeCompare(b.name)
+          : b.name.localeCompare(a.name)
+      ));
+      break;
+
+    default:
+      return sortedProducts;
+  }
+
+  return sortedProducts;
+};
+
+export const Table = ({ products, query, setQuery }) => {
+  // const [query, setQuery] = useState({});
+
+  const sortedProducts = sorter(products, query);
+
+  return (
+    <table
+      data-cy="ProductTable"
+      className="table is-striped is-narrow is-fullwidth"
+    >
+      <thead>
+        <tr>
+          <th>
+            <span className="is-flex is-flex-wrap-nowrap">
+              ID
+
+              <a
+                href="#/"
+                onClick={() => setQuery(() => {
+                  if (query[QUERY.ID] === 'DESC') {
+                    return {};
+                  }
+
+                  return ({
+                    [QUERY.ID]: query[QUERY.ID] ? 'DESC' : 'ASC',
+                  });
+                })}
+              >
+                <span className="icon">
+                  <i
+                    data-cy="SortIcon"
+                    className={cn(
+                      'fas',
+                      {
+                        'fas fa-sort': !query[QUERY.ID],
+                        'fas fa-sort-down': query[QUERY.ID] === 'DESC',
+                        'fas fa-sort-up': query[QUERY.ID] === 'ASC',
+                      },
+                    )}
+                  />
+                </span>
+              </a>
+            </span>
+          </th>
+
+          <th>
+            <span className="is-flex is-flex-wrap-nowrap">
+              Product
+
+              <a
+                href="#/"
+                onClick={() => setQuery(() => {
+                  if (query[QUERY.PRODUCT] === 'DESC') {
+                    return {};
+                  }
+
+                  return ({
+                    [QUERY.PRODUCT]: query[QUERY.PRODUCT] ? 'DESC' : 'ASC',
+                  });
+                })}
```

Comment:
When you cannot write the logic of function in one line it is better to create this function higher in components body